### ISSUE TITLE
Type imports do not follow rules from import/order

### ISSUE THREAD

**Issue by ImRodry** [import-js/eslint-plugin-import#2347]
I have the import/order rule set up with the following configuration:
```json
		"import/order": ["error", {
			"groups": [	"builtin", "index", "external", "internal", "sibling", "parent", "type"],
			"newlines-between": "always",
			"alphabetize": { "order": "asc", "caseInsensitive": true }
		}],
```
And I expected the groups rule to be applied to type imports, however this doesn't seem the case and these imports seem to just get sorted alphabetically. Here's an example:
```ts
// Sorted type imports by import/order
import type { Command } from "../../lib/imports"
import type { WithId } from "mongodb"
```

```ts
Expected sorting following the order external -> internal
import type { WithId } from "mongodb"
import type { Command } from "../../lib/imports"
```

As I understand this may be seen as expected behavior for some people, I believe this fix could be introduced as a boolean option for the rule that would come disabled by default.

**Comment by ljharb** [comment]
They should be sorted by specifier. The import names have no bearing on the sorting.

I definitely agree that it makes sense to, within the group "type", sort everything by `groups` as well. It'd probably be best to do this under an option, yes.

**Comment by ImRodry** [comment]
Ah sorry for my misunderstanding! Is there any documentation on what each one of the groups mean so that I can try to configure my rule better?
Glad to know you agree with this though, looking forward to seeing this in a future release!

**Comment by ljharb** [comment]
I don't think there's any way to configure it for what you want, at the moment. The best path forward is a PR adding the option you want :-)
